### PR TITLE
property service for search api and unit tests 

### PR DESCRIPTION
**Pull request by ericssy**
- added filters in /service/property.py
- added unit test for post_property 
- issues
    * the filter in property service returns correct property objects but the Marshmallow dump() failed to serialize the objets
    * get_property unit test failed

### CODE REVIEW

**Comment by ericssy** [comment]
tested with 

`(venv) MacBook-Pro-49:backend ShenSiyuan$  curl -X POST  -H "Authorization: Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJleHAiOjE1ODgzOTY0MzgsInVpZCI6ImZhM2I1NjI1LWY2Y2UtNDU1Yi1iNTdlLTE2ODk4YThiYjE2YyIsInJvbGUiOiJuYXZpZ2F0b3IifQ.I8MsfuBg4bSOESThzRjH5dOaTc4D5u2-KphIEwXUP-4" -H "accept: application/json" -H "Content-Type: application/json" -d "{"pagination" : {"results_per_page" : 1, "page" : 1}, "searchFields" : {"zip_code" : ["22903", "22904"], "bedrooms" : 1, "date_available" : "2020-12-01"}}" http://localhost:5000/property/search`

`{"pagination":{},"results":[{},{},{}]}`

File: backend/src/property_handlers.py [hunk 1]
```diff
@@ -43,12 +43,20 @@ def get_property():
...
+
         payload = GetPropertyRequest().load(request.get_json())
         _property = app.services.property_service().get_property(payload)
+        print(_property)
+        print(len(_property))
         _property_response = GetPropertyResponse()
-        _property_response.pagination = PaginationResponse()
-        _property_response.results = PropertySchema().dump(_property)
+        _property_response.pagination = PaginationResponse(partial=True).dump(
+            {"TotalNumberOfResults" : len(_property)})
```
Comment (rakshita-kaulgud) [inline backend/src/property_handlers.py:58]: {
                "results_per_page": 10,
                "page": 1,
                "totalNumberOfResults": len(_property)
}

it should be 'totalNumberOfResults'

Comment (ericssy) [reply backend/src/property_handlers.py:58]: thanks!

File: backend/src/property_handlers.py [hunk 2]
```diff
@@ -43,12 +43,20 @@ def get_property():
                         schema: ErrorResponse
     """
     try:
```
Comment (rakshita-kaulgud) [inline backend/src/property_handlers.py:45]: Use this instead:

 try:
        payload = GetPropertyRequest().load(request.get_json())
        _property = app.services.property_service().get_property(payload)

        _property_response = GetPropertyResponse()
        _property_response.pagination = PaginationResponse()
        _property_response.pagination = PaginationResponse(partial=True).load(
            {
                "results_per_page": payload.,
                "page": 1,
                "totalNumberOfResults": len(_property)
            }
        )
        _property_response.results = _property
        return jsonify(GetPropertyResponse().dump(_property_response))

File: backend/src/app/api.py [hunk 3]
```diff
@@ -74,6 +74,6 @@ class Meta:
     load_instance = True
```
Comment (rakshita-kaulgud) [inline backend/src/app/api.py:74]: make the following changes to PropertySchema
class PropertySchema(SQLAlchemyAutoSchema):

**Comment by ericssy** [comment]
fixed the search api and unit tests failure

File: backend/src/property_handlers.py [hunk 4]
```diff
@@ -43,12 +43,18 @@ def get_property():
...
-        print("sent request = " + str(request.get_json()))
         payload = GetPropertyRequest().load(request.get_json())
         _property = app.services.property_service().get_property(payload)
         _property_response = GetPropertyResponse()
         _property_response.pagination = PaginationResponse()
-        _property_response.results = PropertySchema().dump(_property)
+        _property_response.pagination = PaginationResponse(partial=True).load(
+        {
+            "results_per_page": len(_property),
```
Comment (rakshita-kaulgud) [inline backend/src/property_handlers.py:52]: I know we aren't technically implementing pagination right now, but this should contain the values passed in the request

**Pull request merged**